### PR TITLE
fixup regression regarding longtable handling (and some tests fixes)

### LINKED ISSUE #9906: Biblatex related issue caused by Quarto update (probably due to Pandoc update from 3.1.13 to 3.2)
### Bug description

I use quarto to prepare my CV, I just reconfigured quarto based on the latest commit before rendering my CV. However, it throws errors complaining about missing files:

````bash
updating tlmgr

updating existing packages
finding package for numeric.dbx
finding package for biblatex-dm.cfg
finding package for JulianTaoCV.bbl
ERROR: 
compilation failed- no matching packages
Emergency stop.
<*> JulianTaoCV.tex
                    

see /home/julian/Dropbox (ASU)/jtaocv/JulianTaoCV.log for more information.
ERROR: Error
    at renderFiles (file:///home/julian/quarto-cli/src/command/render/render-files.ts:
350:23)
    at eventLoopTick (ext:core/01_core.js:153:7)
    at async renderProject (file:///home/julian/quarto-cli/src/command/render/project.
ts:440:23)
    at async renderForPreview (file:///home/julian/quarto-cli/src/command/preview/prev
iew.ts:428:24)
    at async render (file:///home/julian/quarto-cli/src/command/preview/preview.ts:172
:22)
    at async preview (file:///home/julian/quarto-cli/src/command/preview/preview.ts:18
9:18)
    at async Command.actionHandler (file:///home/julian/quarto-cli/src/command/preview
/cmd.ts:421:7)
    at async Command.execute (file:///home/julian/quarto-cli/src/vendor/deno.land/x/cl
iffy@v1.0.0-rc.3/command/command.ts:1948:7)
    at async Command.parseCommand (file:///home/julian/quarto-cli/src/vendor/deno.land
/x/cliffy@v1.0.0-rc.3/command/command.ts:1780:14)
    at async quarto (file:///home/julian/quarto-cli/src/quarto.ts:159:5)
````

I tried to reinstall TinyTex, cleanup other versions of textlive, updating packages using tlmgr, but none of them worked.

I later reconfigured quarto based on an earlier commit when pandoc 3.1.13 is still used, the rendering was successful and I was able to generate the PDF as before. 

Thus, I suspect that the error was due to the update of Pandoc. 

### Steps to reproduce

1. Fork my CV repo, https://github.com/juliantao/jtaocv
2. It is suggested to use the virtual environment based on the `renv.lock` file. This makes sure that necessary packages are installed and the correct versions of two important packages `RefManageR` and `bibtex` are used
3. configure quarto based on the latest commit, by following the instructions here https://github.com/quarto-dev/quarto-cli
````bash
git clone https://github.com/quarto-dev/quarto-cli
cd quarto-cli
./configure.sh
````
4. Go to the CV folder, render any of the `qmd` files, such as the `JulianTaoCV_1page`. The above errors will appear, and the rendering will be terminated immaturely. 
5. Use an earlier version of quarto, for example, 95579925, reconfigure quarto
````bash
cd quarto-cli
git checkout 95579925
./configure.sh
````
6. then go to the CV folder again, render the same `qmd` file. This should result in successful rendering.

### Expected behavior

_No response_

### Actual behavior

_No response_

### Your environment

_No response_

### Quarto check output

````bash
Quarto 99.9.9
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.13: OK
      Dart Sass version 1.70.0: OK
      Deno version 1.41.0: OK
      Typst version 0.11.0: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 99.9.9
fatal: not a git repository (or any of the parent directories): .git
      Path: /home/julian/quarto-cli/package/dist/bin

[✓] Checking tools....................OK
      TinyTeX: v2024.06
      Chromium: 869685

[✓] Checking LaTeX....................OK
      Using: TinyTex
      Path: /home/julian/.TinyTeX/bin/x86_64-linux
      Version: 2024

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.10.12
      Path: /usr/bin/python3
      Jupyter: 4.10.0
      Kernels: python3

(/) Checking Jupyter engine render....[IPKernelApp] ERROR | No such comm target registered: quarto_kernel_setup
[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........OK
      Version: 4.1.2
      Path: /usr/lib/R
      LibPaths:
        - /home/julian/R/x86_64-pc-linux-gnu-library/4.1
        - /usr/local/lib/R/site-library
        - /usr/lib/R/site-library
        - /usr/lib/R/library
      knitr: 1.43
      rmarkdown: 2.23

[✓] Checking Knitr engine render......OK
````

### PR DESCRIPTION
**Pull request by cderv**
This is a regression following lua regex change

Change happened at c8bfc13 and introduced regression by changing the condition.
This commits correctly put back the right condition

And also adds two tests that would have allow us to find this problem. 

This PR also adds improvement in our test suites by adding a function to test the intermediate .tex file while testing the pdf output rendering with LaTeX too. 

* `ensureLatexFileRegex` is the equivalent of `ensureTypstRegex`. 
	* `keep-tex: true` is required
	* Intermediate .tex file is correctly cleaned up. 
	* Same behavior as with typst.
	 	
* This was done by a refactoring with a new `verifyKeepFileRegexMatches` that could be used in our regular test suite (not smoke all). I moved function around to regroup so diff may not be easy to read. 🤔 

* Some error are also now thrown by the test suite to help find early some problem
	* Using in a smoke-all a verify function that does not exist was previously ignored. 
		* meaning Nothing was tested in case of a typo for example
	* When `ensureLatexFileRegex` or `ensureTypstRegex` are used, respectively `keep-tex: true` and `keep-typ: true` are checked as they are required.
  
Hopefully, all the test will run ok !

### CODE REVIEW

**Comment by cderv** [comment]
All tests are fixed now. We add some non-working tests due to wrong verify function in tests specification. This won't happen. 
I will also check the feature-matrix, against this branch

**Comment by cderv** [comment]
Fixed some issues found in feature-format matrix docs: New run at https://github.com/quarto-dev/quarto-cli/actions/runs/9423620780

File: dev-docs/feature-format-matrix/qmd-files/crossref/float/table/document.qmd [hunk 1]
```diff
@@ -32,6 +32,7 @@ _quarto:
           - "a[href=\"#tbl-2\"].quarto-xref"
     dashboard: *dom-tests
     revealjs: # reveal resolves anchors differently, using the section headings instead of the float ids
+      ensureHtmlElements:
```
Comment (cderv) [inline dev-docs/feature-format-matrix/qmd-files/crossref/float/table/document.qmd:35]: Just a head up that fixing this is breaking the feature format matrix test 
https://github.com/quarto-dev/quarto-cli/actions/runs/9423620780/job/25962387025#step:26:214

So I don't know if revealjs is just not well supported here, or something else but this is probably to fix in another issue. 

This PR just surfaced it as previously there was no test, and it was silently not testing

**Comment by cscheid** [comment]
All the test fixes are great. I wonder if we should fail every test that has a `verify*` key that isn't found...

**Review by cscheid — APPROVED** [review]
LGTM.

All the test fixes indicate that we need to be better at preventing bad tests from being added.

I try to explicitly ensure a test is actually failing before fixing the code, but clearly some of the tests I added would never fail because of bad syntax. So we need stronger checks.

**Comment by cderv** [comment]
> I wonder if we should fail every test that has a verify* key that isn't found...

I used this pattern 
````ts
} else {
            throw new Error(`Unknown verify function used: ${key} in file ${input} for format ${format}`) ;
````

So there will be an _uncaught_ error I guess which will make the test fails already.... 

It is just an unhandled error. Live example here: 
https://github.com/quarto-dev/quarto-cli/actions/runs/9423009019/job/25960520233?pr=9925#step:26:450

**Comment by cscheid** [comment]
> It is just an unhandled error. Live example here: https://github.com/quarto-dev/quarto-cli/actions/runs/9423009019/job/25960520233?pr=9925#step:26:450

Ah, you (of course :)) did the thing I was suggesting it would be a good idea to do! I just missed it in the PR amid the other changes. Greaat!

**Pull request merged**